### PR TITLE
Construction:WindowEquivalentLayer using invalid layer material fatal out with error message

### PR DESCRIPTION
**Pull request by Nigusse**
Pull request overview
---------------------
Addresses issue #7048. Invalid material layers in Construction:WindowEquivalentLayer is checked and if found throws appropriate severe error message before fatal out. Diffs are not expected based on this change.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [x] Unit Test(s)

### CODE REVIEW

File: src/EnergyPlus/HeatBalanceManager.cc [hunk 1]
```diff
@@ -4502,6 +4502,17 @@ namespace HeatBalanceManager {
...
                     ErrorsFound = true;
                 } else {
+                    MaterialLayerGroup = Material(Construct(TotRegConstructs + ConstrNum).LayerPoint(Layer)).Group;
+                    if (!((MaterialLayerGroup == GlassEquivalentLayer) || (MaterialLayerGroup == ShadeEquivalentLayer) ||
+                          (MaterialLayerGroup == DrapeEquivalentLayer) || (MaterialLayerGroup == BlindEquivalentLayer) ||
+                          (MaterialLayerGroup == ScreenEquivalentLayer) || (MaterialLayerGroup == GapEquivalentLayer))) {
+                        ShowSevereError("Invalid material layer type in window " + CurrentModuleObject + " = " +
+                                        Construct(TotRegConstructs + ConstrNum).Name);
+                        ShowContinueError("...Window layer material type = " + ConstructAlphas(Layer) +
+                                          " is not allowed in Construction:WindowEquivalentLayer window object.");
```
Comment (mjwitte) [inline src/EnergyPlus/HeatBalanceManager.cc:4508]: @Nigusse This is good, except `ConstructAlpahs(Layer)` is the material name, not the material type, so the error message is a little confusing:
```
   ** Severe  ** Invalid material layer type in window Construction:WindowEquivalentLayer = SERRAMENTI PVC FINESTRE DOPPIA CAMERA 7CM
   **   ~~~   ** ...Window layer material type = VETRO DOPPIA CAMERA 4-9-4 NON TRATTATO SUPERFIC. is not allowed in Construction:WindowEquivalentLayer window object.
```
Maybe just delete "material type" above. And add "Only materials of type Material:*:EquivalentLayer are allowed."

**Review by mjwitte — COMMENTED** [review]
Tested this with modified defect file and it now traps the illegal material with a meaningful error message. Ran unit tests - all OK. Please fix the error message as noted below and merge in develop and then this can go in once CI comes back all green.

File: src/EnergyPlus/HeatBalanceManager.cc (hunk 1 above)
Comment (rraustad) [reply src/EnergyPlus/HeatBalanceManager.cc:4508]: But then the user has to identify which material is wrong. Small effort I know but wouldn't it be better to point to it?

Comment (mjwitte) [reply src/EnergyPlus/HeatBalanceManager.cc:4508]: My suggestion was to keep the name, just delete two words:
** ... Window layer ~material type~ = VETRO VETRO DOPPIA CAMERA 4-9-4 NON TRATTATO SUPERFIC. is not allowed ....
**  Only materials of type Material:*:EquivalentLayer are allowed.

**Comment by Myoldmopar** [comment]
OK, I updated the error message per @mjwitte request.  I took the defect file and ran it with both develop and this branch.  As expected, with the degree symbol it fails in both with an incorrectly "blank" name.  When I clean up the object name, on develop the results go wild, while on this branch I get a nice error message:

```
   ** Severe  ** Invalid material layer type in window Construction:WindowEquivalentLayer = SERRAMENTI PVC FINESTRE DOPPIA CAMERA 7CM
   **   ~~~   ** ...Window layer = VETRO DOPPIA CAMERA 4-9-4 NON TRATTATO SUPERFIC. is not allowed in Construction:WindowEquivalentLayer window object.
   **   ~~~   ** Only materials of type Material:*:EquivalentLayer are allowed
   **  Fatal  ** GetSurfaceData: Errors discovered, program terminates.
```

I'm pushing up my changes and merging in.

**Pull request merged**